src/NHibernate/Type/AbstractStringType.cs

```diff
@@ -1,13 +1,42 @@
 using System;
+using System.Collections.Generic;
 using System.Data.Common;
+using System.Globalization;
 using NHibernate.Engine;
 using NHibernate.SqlTypes;
+using NHibernate.UserTypes;
 
 namespace NHibernate.Type
 {
 	[Serializable]
-	public abstract class AbstractStringType: ImmutableType, IDiscriminatorType
+	public abstract class AbstractStringType: ImmutableType, IDiscriminatorType, IParameterizedType
 	{
+		/// <summary>
+		/// The comparer culture parameter name. Value should be <c>Current</c>, <c>Invariant</c>,
+		/// <c>Ordinal</c> or any valid culture name.
+		/// </summary>
+		/// <remarks>Default comparison is ordinal.</remarks>
+		public const string ComparerCultureParameterName = "ComparerCulture";
+
+		/// <summary>
+		/// The case sensitivity parameter name. Value should be a boolean, <c>false</c> meaning
+		/// case insensitive.
+		/// </summary>
+		/// <remarks>Default comparison is case sensitive.</remarks>
+		public const string CaseSensitiveParameterName = "CaseSensitive";
```

Comment:
`CaseSensitive` is a bad parameter because its default value is `true` which contradicts the default value of `Boolean`, so it can be confusing.

Reply:
And it does not play nice with the behavior of `boolean.TryParse`, if the culture is set.